File: src/domain/entities/seller.ts
import type { Inventory } from "./inventory.js";

export interface Seller {
	id_seller: number;
	seller_name: string;
	seller_email: string;
	seller_password: string;
	phone: number;
	bussines_name: string;
	location: string;
	created_at: Date;
	inventories?: Inventory[];
}

File: backend/realtime_service/internal/models/seller.go
package models

import "time"

// Seller representa un vendedor simplificado para eventos en tiempo real.
// Sincronizado con SellerEntity del rest_service.
type Seller struct {
	IDSeller     int       `json:"id_seller"`
	SellerName   string    `json:"seller_name"`
	SellerEmail  string    `json:"seller_email"`
	Phone        int64     `json:"phone"`
	BusinessName string    `json:"bussines_name"` // Mantiene el typo del modelo original
	Location     string    `json:"location"`
	CreatedAt    time.Time `json:"created_at"`
}
